server/plugin.go

```diff
@@ -430,11 +438,11 @@ func (p *Plugin) generatePluginSecrets() error {
 	return nil
 }
 
-func (p *Plugin) OnActivate() error {
+func (p *Plugin) OnActivate() (err error) {
```

Comment:
nit: I don't use named return values often, but wondering if it makes sense to do something like `retErr` and /not/ restructure all the other error handling below. Thea idea would be to only use `retErr` inside `defer` callbacks and otherwise /avoid/ named return values. Thoughts?

Reply:
I think I going to move it to an internal `onActivate` function and do the error handling outside, I think is going to be cleaner.